### PR TITLE
Build fixes, especially around cmake+monolithic

### PR DESCRIPTION
**Pull request by jbfove**
Was integrating wxWidgets into another larger cmake project, using the monolithic build, and ran into a number of build issues. Though not all of them are necessarily related to the monolithic build

Contributing back the fixes back so the next person doesn't need to solve them all over again.

No tests are added since these are build fixes, no new or changed features.

This is my first MR for wx so I'm more than happy to rework things or make any changes as needed.

### CODE REVIEW

File: build/cmake/files.cmake [hunk 1]
```diff
@@ -588,7 +588,6 @@ set(BASE_CMN_HDR
     wx/textbuf.h
     wx/textfile.h
     wx/thread.h
-    wx/thrimpl.cpp
```
Comment (MaartenBent) [inline build/cmake/files.cmake]: `files.cmake` is auto-generated, so the build error caused by this file should be solved in another way.
And preferable in it's own commit.

**Review by MaartenBent — COMMENTED** [review]
Looks great.

Monolithic build now works fine in VS 2017. But it takes ages to build because pch are not enabled. This could be fixed in `build\cmake\lib\CMakeLists.txt` with:

```diff
@@ -87,6 +92,7 @@ if(wxBUILD_MONOLITHIC)
             target_compile_definitions(mono ${vis} ${wxMONO_DEFINITIONS_${vis}})
         endif()
     endforeach()
+    wx_finalize_lib(mono)
 endif()
 
 # Propagate variable(s) to parent scope
```

There is also a ticket for this problem: https://trac.wxwidgets.org/ticket/18074

File: build/cmake/files.cmake (hunk 1 above)
Comment (jbfove) [reply build/cmake/files.cmake]: Thanks, I missed the autogenerated note at the top. I'll take a stab at fixing this another way, or remove it from the MR if I can't easily find a fix.

Comment (jbfove) [reply build/cmake/files.cmake]: Found the root issue here: `set_source_files_properties(... HEADER_FILE_ONLY TRUE)` is used to make the .cpp "headers" not compile. However the `mono` target is created in a different scope so it couldn't read the properties set. I introduced a new variable `wxMONO_NONCOMPILED_CPP_FILES` to pass these upscope so the `mono` target can set this within it's own scope. Fix is pushed.

**Comment by jbfove** [comment]
Force pushed 3 changes:
- Proper fix for the files.cmake issue (per comment above)
- Fixed samples building against the mono target(by using `wx_exe_link_libraries` instead of `target_link_libraries`)
- Added the trac number to the commit message (note: all my monolithic-related fixes are in one commit since it doesn't work without all of them, but I can split them into individual commits, where only the last one will work, if that's preferred)
- Added `wx_finalize_lib(mono)` per your suggestion, speed up the windows builds (but currently on OSX so I haven't tested this yet)

**Comment by vadz** [comment]
Thanks for the fixes! I can't meaningfully review the CMake part, but Maarten has already review that, and the rest looks good to me, so merged now.

Thanks again!

**Comment by LegalizeAdulthood** [comment]
CMake part looks good to me.

**Pull request merged**